frontend/src/components/Form.tsx

```diff
@@ -0,0 +1,162 @@
+import { Tabs, TabList, Tab, TabPanels, TabPanel } from "@reach/tabs";
+import { useRouter } from "next/router";
+import { ChangeEvent, useContext, useEffect, useState } from "react";
+import { useTranslations } from "next-intl";
+import getImage from "../utils/getImage";
+import Tag from "./Tag";
+import InputRadio from "./InputRadio";
+import Player from "./Player";
+import Button from "./Button";
+import bbcode from "../libs/bbcode";
+import UserContext from "../context/UserContext";
+import { Answers, PostsId, UserInAnswer } from "../types/common-types";
+import { dynamicSort } from "../utils/objectSort";
+
+const TempPlayers = [
+  { name: "Varvalian", ranking: 14, countryRanking: 1, discordTag: "Varvalian#948" },
+  { name: "Alumetri", ranking: 47, countryRanking: 3, discordTag: "Alumetri#836" },
+  { name: "sakamata1", ranking: 1, countryRanking: 1, discordTag: "sakamata1#1337" },
+  { name: "badeu", ranking: 38, countryRanking: 1, discordTag: "badeu#1114" },
+  { name: "WhiteCat", ranking: 4, countryRanking: 1, discordTag: "WhiteCat#1076" }
+];
+
+interface IFromProps {
+  posts: PostsId;
+  users: UserInAnswer[] | null | undefined;
+  answers: Answers;
+}
+
+export default function Form({ posts, users, answers }: IFromProps) {
+  const { banner, description, icon, id, is_active, title } = posts;
+  const { user } = useContext(UserContext);
+  const osuUser = user?.osu;
+
+  const [sort, setSort] = useState("rank");
+  const [sortedPlayers, setSortedPlayers] = useState(TempPlayers);
+
+  const [primaryAuthor, setPrimaryAuthor] = useState<UserInAnswer | null | undefined>(null);
```

Comment:
detail: we should definitely set a default value here instead of null since it is server rendered and it won't have the author for first render